swan-lake/integration-technology-tools/edi-tool.md

```diff
@@ -6,4 +6,349 @@ keywords: ballerina, programming language, edi, contract
+    "segments" : {
+        "HDR": {
+            "tag" : "header",
+            "fields" : [{"tag" : "code"}, {"tag" : "orderId"}, {"tag" : "organization"}, {"tag" : "date"}]
+        },
+        "ITM": {
+            "tag" : "items",
+            "maxOccurances" : -1,
+            "fields" : [{"tag" : "code"}, {"tag" : "item"}, {"tag" : "quantity", "dataType" : "int"}]
+        }
+    }
+}
+```
+
+The above schema can be used to parse EDI documents with one HDR segment (mapped to `header`) and any number of ITM segments (mapped to `items`). The HDR segment contains three fields, which are mapped to `orderId`, `organization`, and `date`. Each ITM segment contains two fields mapped to `item` and `quantity`. Below is a sample EDI document that can be parsed using the above schema (`edi-sample1.edi` file).
+
+```
+HDR*ORDER_1201*ABC_Store*2008-01-01~
+ITM*A-250*12
+ITM*A-45*100
+ITM*D-10*58
+ITM*K-80*250
+ITM*T-46*28
+```
+
+### Generate Ballerina records
+
+Ballerina records for the EDI schema in the `edi-schema.json` file can be generated as follows (the generated Ballerina records will be saved in `orderRecords.bal`).
+
+```
+bal edi codegen -s resources/edi-schema1.json -o modules/hmartOrder/orderRecords.bal
+```
+
+The Ballerina records generated for the above schema are shown below.
+
+```ballerina
+type Header_Type record {|
+   string code?;
+   string orderId?;
+   string organization?;
+   string date?;
+|};
+
+type Items_Type record {|
+   string code?;
+   string item?;
+   int quantity?;
+|};
+
+type SimpleOrder record {|
+   Header_Type header;
+   Items_Type[] items?;
+|};
+```
+
+### Read EDI files
+
+The generated `fromEdiString` function can be used to read EDI text files into the generated Ballerina record as shown below. 
+
+>**Note:** Any data item in the EDI can be accessed using the record's fields as shown in the example code below.
+
+```ballerina
+import ballerina/io;
+
+public function main() returns error? {
+    string ediText = check io:fileReadString("resources/edi-sample1.edi");
+    SimpleOrder order1 = check hmartOrder:fromEdiString(ediText);
+    io:println(order1.header.date);
+}
+```
+
+### Write EDI files
+
+The generated `toEdiString` function can be used to serialize the `SimpleOrder` records into EDI text as shown below.
+
+```ballerina
+import ballerina/io;
+
+public function main() returns error? {
+    SimpleOrder order2 = {...};
+    string orderEDI = check hmartOrder:toEdiString(order2);
+    io:println(orderEDI);
+}
+```
+
+## Create an EDI library
+
+Usually, organizations have to work with many EDI formats, and integration developers need to have a convenient way to work on EDI data with minimum effort. Ballerina EDI libraries facilitate this by allowing organizations to pack all EDI processing the codes of their EDI collections into an importable library. Therefore, integration developers can simply import those libraries and convert EDI messages into Ballerina records in a single line of code.
```

Comment:
Incorrect terminology in `libraries`

Ballerina has packages. Collection of packages is a `library`. So there is nothing as `libraries`

Reply:
Fixed.